server.js

```diff
@@ -22,7 +23,38 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// Start defining your RESTful end points/routes here
+app.get('/catalogue', (req, res) => {
+  res.json(netflixData)
+})
+
+app.get('/type/:type', (req, res) => {
+  let type = req.params.type.toLowerCase().replace(/[-_ ]/, "")
+  if (type === "tvshow" || type === "show" || type === "tv") {
+    type = "TV Show"
+  } else if (type === "movie") {
+    type = "Movie"
+  }
+  const catalogueType = netflixData.filter((item) => item.type === type)
+  res.json(catalogueType)
+})
+app.get('/year/:year', (req, res) => {
+  const year = req.params.year
+  let type = req.query.type
+  let catalogueYear = netflixData.filter((item) => item.release_year === +year)
+
+  if (type) {
+    type = type.toLowerCase().replace(/[-_]/, "")
+    if (type === "tvshow" || type === "show" || type === "tv") {
+      type = "TV Show"
+    } else if (type === "movie") {
+      type = "Movie"
+    }
+    catalogueYear = catalogueYear.filter((item) => item.type === type)
```

Comment:
Just an idea, since this logic is the same as the one in the other endpoint, but using different data, you could abstract the logic a bit to make it reusable. For example, if you made a function like this:

```
const dataByType = (unformattedType, data) => {
  type = unformattedType.toLowerCase().replace(/[-_]/, "")
  if (type === "tvshow" || type === "show" || type === "tv") {
    type = "TV Show"
  } else if (type === "movie") {
    type = "Movie"
  }

  return data.filter((item) => item.type === type)
}
```

Then you could use it in this endpoint:

```
if (type) {
   res.json(dataByType(type, catalogueYear))
} else {
  res.json(catalogueYear)
}
```

And the same approach in the `/type/:type` endpoint above:

```
res.json(dataByType(type, netflixData))
```